src/lib/comms/include/sol-coap.h

```diff
@@ -141,93 +198,469 @@ typedef enum {
+     * @param resource The resource the request mas made on.
+     * @param req Packet containing the request data. It's not safe to keep a
+     *            reference to it after this function returns.
+     * @param cliaddr The source address of the request.
+     * @param data User data pointer provided to sol_coap_server_register_resource().
+     *
+     * @return 0 on success, -errno on failure.
      */
     int (*get)(struct sol_coap_server *server,
         const struct sol_coap_resource *resource,
         struct sol_coap_packet *req,
         const struct sol_network_link_addr *cliaddr, void *data);
+    /**
+     * POST request.
+     *
+     * @param server The server through which the request was made.
+     * @param resource The resource the request mas made on.
+     * @param req Packet containing the request data. It's not safe to keep a
+     *            reference to it after this function returns.
+     * @param cliaddr The source address of the request.
+     * @param data User data pointer provided to sol_coap_server_register_resource().
+     *
+     * @return 0 on success, -errno on failure.
+     */
     int (*post)(struct sol_coap_server *server,
         const struct sol_coap_resource *resource,
         struct sol_coap_packet *req,
         const struct sol_network_link_addr *cliaddr, void *data);
+    /**
+     * PUT request.
+     *
+     * @param server The server through which the request was made.
+     * @param resource The resource the request mas made on.
+     * @param req Packet containing the request data. It's not safe to keep a
+     *            reference to it after this function returns.
+     * @param cliaddr The source address of the request.
+     * @param data User data pointer provided to sol_coap_server_register_resource().
+     *
+     * @return 0 on success, -errno on failure.
+     */
     int (*put)(struct sol_coap_server *server,
         const struct sol_coap_resource *resource,
         struct sol_coap_packet *req,
         const struct sol_network_link_addr *cliaddr, void *data);
+    /**
+     * DELETE request.
+     *
+     * @param server The server through which the request was made.
+     * @param resource The resource the request mas made on.
+     * @param req Packet containing the request data. It's not safe to keep a
+     *            reference to it after this function returns.
+     * @param cliaddr The source address of the request.
+     * @param data User data pointer provided to sol_coap_server_register_resource().
+     *
+     * @return 0 on success, -errno on failure.
+     */
     int (*delete)(struct sol_coap_server *server,
         const struct sol_coap_resource *resource,
         struct sol_coap_packet *req,
         const struct sol_network_link_addr *cliaddr, void *data);
+    /**
+     * Bitwise OR-ed flags from #sol_coap_flags, if any is necessary.
+     */
     enum sol_coap_flags flags;
     struct sol_str_slice iface;
```

Comment:
For those following at home, both `iface` and `resource_type` are deprecated and will be nuked soon.